nixos/modules/services/x11/xbanish.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.xbanish;
+
+in {
+  options.services.xbanish = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable xbanish to hide your mouse pointer while not in use";
+    };
+
+    arguments = mkOption {
+      description = "Arguments to pass to xbanish command";
+      default = "";
+      example = "-d -i shift";
+      type = types.str;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.xbanish = {
+      description = "xbanish";
+      wantedBy = [ "default.target" ];
```

Comment:
Upstream recommends linking `default.target` to `multi-user.target` or `graphical.target` depending on system type. I think on nixos we always link to `multi-user.target` and something pulls in `graphical.target`
So it probably does not make much of a difference, but in my opinion it's cleaner to explicitly use the more descriptive (and intended) target.
Also keep in mind what happens when `graphical.target` is for example restarted: the restart should be propagated which will not happen unless we set the dependency right.